src/arcaflow_plugin_sdk/schema.py

```diff
@@ -2636,21 +2695,70 @@ class OneOfStringSchema(_JSONSchemaGenerator, _OpenAPIGenerator):
     ...                 )
     ...             }
     ...         )
-    ...     }
+    ...     },
+    ...     discriminator_inlined=False,
     ... )
     """  # noqa: E501
 
     types: Dict[str, typing.Annotated[_OBJECT_LIKE, discriminator("type_id")]]
+    discriminator_inlined: typing.Annotated[
+        bool,
+        _name("Discriminator field inlined"),
+        _description(
+            "Whether or not the discriminator is inlined in the underlying"
+            " objects' schema"
+        ),
+    ]
     discriminator_field_name: typing.Annotated[
         str,
         _name("Discriminator field name"),
         _description(
-            "Name of the field used to discriminate between possible values."
-            " If this field ispresent on any of the component objects it must"
-            " also be a string."
+            "Name of the field whose value is used to discriminate between"
+            " possible subobject types. If this field is present in any of the"
+            " subobjects it must have a type of string."
         ),
     ] = "_type"
 
+    def _insert_discriminator(
+        self,
+        discriminated_object: typing.Dict[str, typing.Any],
+        discriminator_val: str,
+    ) -> typing.Dict[str, typing.Any]:
+        """Add a discriminator field as a property of a member type.
+
+        This function adds a member type's discriminator field as a property
+        with a constant value equal to its discriminated value. The
+        discriminator field is moved to the zeroth index of the list of
+        required fields in a data packet.
+
+        :param discriminated_object: A dictionary in the form of a json
+            fragment of an object's schema in the scope of this OneOf's union.
```

Comment:
This description feels inside-out:  `discriminated_object` is the schema for one of the objects in the OneOf's union (right?); it is provided as a Python `dict` which represents the relevant fragment of the scope's JSON definition.

But, it's not obvious to me that it is important, in the context of this function, that `discriminated_object` is a fragment of some other schema, and it really doesn't matter that that schema was once represented in JSON.

What matters is that we've got a schema for an object (in `dict` form) and that we intend to modify it.

Note that Python passes objects by reference, so we are modifying the object to which `discriminated_object` refers and then returning a new reference to that same object...which makes returning a value possibly superfluous, since the caller already has this reference (because they passed it into this function).

Any change made here should be reflected in the `:returns:` section, as well.   And, in the other instance of the `_insert_discriminator()` function, too....

Reply:
I added the change in description of the `discriminated_object` parameter.